crates/sui/src/client_commands.rs

```diff
@@ -1786,6 +1767,52 @@ pub struct NewAddressOutput {
     pub recovery_phrase: String,
 }
 
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ObjectsOutput {
+    pub object_id: ObjectID,
+    pub version: SequenceNumber,
+    pub digest: String,
+    pub owner_type: String,
+    pub object_type: String,
+}
+
+impl ObjectsOutput {
+    fn from(obj: SuiObjectResponse) -> Result<Self, anyhow::Error> {
+        let obj = obj.into_object()?;
+        let owner_type = match obj.owner {
+            Some(Owner::AddressOwner(_)) => "AddressOwner",
+            Some(Owner::ObjectOwner(_)) => "ObjectOwner",
+            Some(Owner::Shared { .. }) => "Shared",
+            Some(Owner::Immutable) => "Immutable",
+            None => "None",
+        }
+        .to_string();
+        // this replicates the object type display as in the sui explorer
+        let object_type = match obj.type_ {
+            Some(sui_types::base_types::ObjectType::Struct(x)) => {
+                let address = x.address().to_string();
```

Comment:
Does this always produce a string of length 64?  It's an extreme edge case, but you should make sure it does, because otherwise if the address is somehow less than 8 characters long, your truncated form will mis-represent it.

Reply:
Thanks for this, I haven't considered that possibility. I'll add a check!